### PR TITLE
feat: implement storage_system_test example

### LINKED ISSUE #100: Implement example: functions_storage_system_test
Line numbers are approximate:

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/functions/helloworld/hello_cloud_storage_system_test.go#L20-L105

https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/functions/helloworld/sample_storage_test_system.py#L15-L70

### PR DESCRIPTION
**Pull request by coryan**
Fixes #100

### CODE REVIEW

**Comment by coryan** [comment]
I do not know how to avoid the code coverage hit, so we will have to take it.

File: examples/site/testing_storage/README.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  BOOST_LOG_TRIVIAL(info) << "Bucket: " << payload.value("bucket", "");
+  BOOST_LOG_TRIVIAL(info) << "Object: " << payload.value("name", "");
+  BOOST_LOG_TRIVIAL(info) << "Metageneration: "
+                          << payload.value("metageneration", "");
+  BOOST_LOG_TRIVIAL(info) << "Created: " << payload.value("timeCreated", "");
+  BOOST_LOG_TRIVIAL(info) << "Updated: " << payload.value("updated", "");
+}
+```
+
+This test receives storage events and lots a few of the fields in these
```
Comment (devbww) [inline examples/site/testing_storage/README.md:52]: s/lots/logs/

File: examples/site/testing_storage/README.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+events.
+
+## Writing a Unit Test
+
+This function uses the [Boost.Log][boost-log-gh] library to facilitate unit
+testing. Logging can be captured and examined using the usual testing
+assertions, see [storage_unit_test.cc] for more details.
```
Comment (devbww) [inline examples/site/testing_storage/README.md:59]: s/, see/. See/

File: examples/site/testing_storage/README.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+```
+
+If needed, create a Google Cloud Storage bucket for these tests:
+
+```shell
+BUCKET_NAME=... # assign a valid bucket name
+gsutil mb -p "${GOOGLE_CLOUD_PROJECT}" "gs://${BUCKET_NAME}"
+```
+
+Then change the bucket to publish all changes to a Cloud Pub/Sub topic, this
```
Comment (devbww) [inline examples/site/testing_storage/README.md:104]: s/, this/. This/

**Review by devbww — APPROVED** [review]

File: examples/site/testing_storage/README.md (hunk 3 above)
Comment (coryan) [reply examples/site/testing_storage/README.md]: Fixed, thanks.

File: examples/site/testing_storage/README.md (hunk 2 above)
Comment (coryan) [reply examples/site/testing_storage/README.md]: Fixed.

File: examples/site/testing_storage/README.md (hunk 1 above)
Comment (coryan) [reply examples/site/testing_storage/README.md]: Fixed.

**Pull request merged**